rosidl_generator_cpp/resource/msg__meta.hpp.template

```diff
@@ -0,0 +1,77 @@
+// generated from rosidl_generator_cpp/resource/msg__meta.hpp.template
+
+@#######################################################################
+@# EmPy template for generating <msg>__meta.hpp files
+@#
+@# Context:
+@#  - spec (rosidl_parser.MessageSpecification)
+@#    Parsed specification of the .msg file
+@#  - subfolder (string)
+@#    The subfolder / subnamespace of the message
+@#    Either 'msg' or 'srv'
+@#  - get_header_filename_from_msg_name (function)
+@#######################################################################
+@
+@{
+from rosidl_generator_cpp import MSG_TYPE_TO_CPP
+
+cpp_namespace = '%s::%s::' % (spec.base_type.pkg_name, subfolder)
+
+cpp_bounded_types = ['bool', 'char', 'uint8_t', 'int8_t', 'uint16_t', 'int16_t', 'uint32_t', 'int32_t', 'uint64_t', 'int64_t', 'float', 'double']
+}@
+#include <stdint.h>
+#include <array>
+#include <string>
+#include <vector>
+
+#ifndef __ALL_MSG__BASIC_TEMPLATES
+#define __ALL_MSG__BASIC_TEMPLATES
+
+  template<typename T>
+  struct has_bounded_size { };
+
+@[for bounded_type in cpp_bounded_types]@
+  template<>
+  struct has_bounded_size<@(bounded_type)> { static const bool value = true; };
+@[end for]@
+
+  // TODO(jacquelinekay): for now, all dynamically sized arrays and strings are considered unbounded
+  template<>
+  struct has_bounded_size<std::string> { static const bool value = false; };
+
+  template<typename T>
+  struct has_bounded_size<std::vector<T>> { static const bool value = false; };
+
+#endif  /* __ALL_MSG__BASIC_TEMPLATES */
```

Comment:
Should this block be moved to a separate header file rather then embedded in every generated message?